include/fdreadoutlibs/wib2/tpg/ProcessingInfo.hpp

```diff
@@ -86,19 +96,47 @@ struct ProcessingInfo
     , absTimeModNTAPS(absTimeModNTAPS_)
   {}
 
+
   // Set the initial state from the window starting at first_msg_p
   template<size_t N>
   void setState(const RegisterArray<N>& first_tick_registers)
   {
     static_assert(N >= NREGISTERS, "Wrong array size");
-    // Set the pedestals and the 25/75-percentiles: everything else is just the same as in default values
+
+    // AAA: Loop through all the registers, loop through all the channels, look at the 
+    // first message of the superchunk and read the ADC value. This will be used as the 
+    // pedestal for the channel state
     for (size_t j = 0; j < NREGISTERS * SAMPLES_PER_REGISTER; ++j) {
-      const int16_t ped = first_tick_registers.uint16(j); // NOLINT
+      const size_t register_offset = j % SAMPLES_PER_REGISTER; 
+      const size_t register_index = j / SAMPLES_PER_REGISTER;
+      const size_t register_t0_start = register_index * SAMPLES_PER_REGISTER * FRAMES_PER_MSG;
+
+      int16_t ped;
+      for (size_t itime = 0; itime < timeWindowNumFrames; ++itime) {
+        const size_t msg_index = itime / 12;
+        const size_t msg_time_offset = itime % 12;
+        // The index in uint16_t of the start of the message we want. 
+        // AAA: 6144 is the sizeof(MessageADCs) which is hardcoded here. TODO: put
+        // MessageADCs in a location that can be included and used in the ProcessingInfo 
+        const size_t msg_start_index = msg_index * swtpg_wib2::ADCS_SIZE / sizeof(uint16_t); // NOLINT
+        const size_t offset_within_msg = register_t0_start + SAMPLES_PER_REGISTER * msg_time_offset + register_offset;
+        const size_t index = msg_start_index + offset_within_msg;
+
+        const uint16_t* input16 = first_tick_registers.data(); // NOLINT
+
+        ped = input16[index];
+
+        break; // breaking in order to select only the first entry
+      }
+
+      // Set the pedestals and the 25/75-percentiles: everything else is just the same as in default values
       chanState.pedestals[j] = ped;
-      chanState.quantile25[j] = ped - 3;
-      chanState.quantile75[j] = ped + 3;
+      chanState.pedestalsRS[j] = 0;
+      chanState.RS[j] = 0;
+      chanState.quantile25[j] = ped-20;
```

Comment:
What is the -20? Please expand comments about ops. This is very contrary to understand what is happening via the surrodinging member fields and comments? (qiantile25 but ped is substracted by 20. Also comments are talking about 25/75 percentile?